### PR TITLE
[WIP ] Add AnimateDiff SparseCntrl

### PR DESCRIPTION
**Pull request by DN6**
# What does this PR do?
Add AnimateDiff SparseCntrl model to diffusers 
https://github.com/guoyww/AnimateDiff/tree/main?tab=readme-ov-file#202312-animatediff-v3-and-sparsectrl

TODO:
- [x] Convert checkpoints to diffusers format
- [x] Add sparse control pipeline and sparse controlnet model
- [ ] Docs
- [ ] Ensure outputs are equivalent for original model and diffusers version

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [ ] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/diffusers/pipelines/animatediff/convert_motion_module_ckpt_to_diffusers.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        state_dict = state_dict["state_dict"]
+
+    conv_state_dict = convert_motion_module(state_dict)
+    adapter = MotionAdapter(
+        use_motion_mid_block=args.use_motion_mid_block, motion_max_seq_length=args.motion_max_seq_length
+    )
+    # skip loading position embeddings
+    adapter.load_state_dict(conv_state_dict, strict=False)
+    adapter.save_pretrained(args.output_path)
+    adapter.save_pretrained(args.output_path, variant="fp16")
```
Comment (sayakpaul) [inline src/diffusers/pipelines/animatediff/convert_motion_module_ckpt_to_diffusers.py:51]: For fp16 variant, we should also set the dtype to `torch.float16`. I don't see that being used. This issue persists here too: https://huggingface.co/guoyww/animatediff-motion-adapter-v1-5-2/tree/main. 

Both the fp16 and non-fp16 variants have the same size which shouldn't be the case.

Am I missing out on anything?

**Comment by chuck-ma** [comment]
It seems that AnimateDiffSparseControlNetPipeline cannot be imported because nothing added in __init__.py.
Besides,  the quality seems not good enough. The scribble control model does not effectively control the frames of the corresponding video based on the input controlnet image

**Comment by sayakpaul** [comment]
@DN6

**Comment by a-r-r-o-w** [comment]
@DN6 Hey, what's holding this from merge? Happy to help if there's anything I can look into

**Comment by DN6** [comment]
Hi @a-r-r-o-w sorry been a bit caught up with some other things. Feel free to take over this PR or open a new one if you prefer. 

The SparseControlNet needs to be cleaned up and we need to verify that the outputs match the original models.

**Comment by eps696** [comment]
hello, is this WIP still in progress? 
seems stuck for half a year, which is very sad, as this is a much wanted feature.

i've tried to employ it as is and got some working controlnet-related results, but it worked per-frame (as usual non-sparse controlnet), not affecting the motion. that is, "controlled" (indexed) frames reproduced controlnet inputs while the rest was generated as if there was no controlnet at all. 
on my understanding it looked like the motion part of the model (some 3d blocks?) were not properly connected to the controlnet somehow, or i'm missing something..

**Comment by a-r-r-o-w** [comment]
hello @eps696! we recently shipped SparseCtrl in #8897 and it will be part of the next release. apologies for the delay :) feel free to try it out and let us know if you find any bugs, thanks!

@DN6 this can be closed now, no?

**Comment by DN6** [comment]
Closing in favour of https://github.com/huggingface/diffusers/pull/8897

**Comment by eps696** [comment]
@a-r-r-o-w thanks for the response!
alas it still does not seem to handle properly the input image, giving similar outputs no matter what.
the code is copied from "RGB single image" here https://github.com/huggingface/diffusers/pull/8897#issuecomment-2237767626 , with latest diffusers from the github main branch

animatediff_sparsectrl_rgb_single
animatediff_sparsectrl_rgb_single-2

**Pull request closed without merging**